services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/RouteOptions.java

```diff
@@ -2019,6 +2040,27 @@ public Builder snappingIncludeClosuresList(@Nullable List<Boolean> snappingClosu
     @NonNull
     public abstract Builder experimental(@Nullable Experimental experimental);
 
+    /**
+     * Use this method to add custom request parameters to an url generated by `RouteOptions#toUrl`.
+     * Custom parameters are not present in model yet but supported on the Directions API,
+     * for example experimental parameters.
+     * @param unrecognizedProperties parameters to add to request
+     */
+    @NonNull
+    public Builder unrecognizedProperties(@Nullable Map<String, String> unrecognizedProperties) {
```

Comment:
```suggestion
    public Builder unrecognizedProperties(@Nullable Map<String, JsonPrimitive> unrecognizedProperties) {
```

Reply:
do you think that 
```kotlin
unrecognizedProperties(mapof(
    "test" to JsonPrimitive(4)
))
```
is more convenient than
```kotlin
unrecognizedProperties(mapof(
    "test" to "4"
))
```
?

Reply:
I think that:
```
unrecognizedProperties(mapof(
    "test" to JsonPrimitive(myNumber)
))
```
is clearer than:
```

unrecognizedProperties(mapof(
    "test" to String.valueOf(myNumber)
))
```
because that's how it will look like in practice. Since we already work on `JsonPrimitive`s under the hood, and `getRecognizedProperty` also returns a `JsonElement`, the proposal looks more natural to me.